### PR TITLE
Check LateInitialize management policy in Plugin Framework external client

### PR DESCRIPTION
**Pull request by mergenci**
### Description of your changes

1. Before calling LateInitialize(), make sure that spec.managementPolicies include LateInitialize policy.
2. Extend https://github.com/crossplane/upjet/pull/341 to handle optional attributes, in addition to computed ones.

I have:

- [x] Read and followed Upjet's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

I manually tested using an AWS Security Group Ingress Rule with the following configuration:

```
apiVersion: ec2.aws.upbound.io/v1beta1
kind: VPC
metadata:
  annotations:
    meta.upbound.io/example-id: ec2/v1beta1/securitygroupingressrule
  name: test-pr-350-vpc
  labels:
    testing.upbound.io/example-name: test-pr-350-vpc
spec:
  forProvider:
    region: us-west-1
    cidrBlock: 172.16.0.0/16
    tags:
      Name: TestCemVPC

---
apiVersion: ec2.aws.upbound.io/v1beta1
kind: SecurityGroup
metadata:
  annotations:
    meta.upbound.io/example-id: ec2/v1beta1/securitygroupingressrule
  name: test-pr-350-securitygroup
  labels:
    testing.upbound.io/example-name: test-pr-350-securitygroup
spec:
  forProvider:
    region: us-west-1
    vpcIdSelector:
      matchLabels:
        testing.upbound.io/example-name: test-pr-350-vpc

---
apiVersion: ec2.aws.upbound.io/v1beta1
kind: SecurityGroupIngressRule
metadata:
  name: test-pr-350-securitygroupingressrule
spec:
  managementPolicies: ["Create", "Observe", "Update", "Delete", "LateInitialize"]
  initProvider:
    tags:
      foo: barInitProvider
  forProvider:
    cidrIpv4: 10.0.0.0/8
    fromPort: 8080
    ipProtocol: tcp
    region: us-west-1
    securityGroupIdRef:
      name: test-pr-350-securitygroup
    toPort: 8081
```

[`description` field is optional](https://github.com/upbound/terraform-provider-aws/blob/ace568ed3f1f82e3ff00cf2bf8c243e39d12b435/internal/service/ec2/vpc_security_group_ingress_rule.go#L117-L119) and is not specified in the configuration above. After the resource is created, set its description from AWS Console and trigger a reconciliation. Check that `spec.forProvider.description` and `status.atProvider.description` are set after the reconciliation is finished.

### References

This PR is a follow-up to https://github.com/crossplane/upjet/pull/343.

[contribution process]: https://github.com/crossplane/upjet/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Comment by mergenci** [comment]
@erhancagirici, do you have any concerns over [this modification](https://github.com/crossplane/upjet/pull/350/files#diff-a0fac6561a445dda5c6890258d64064e814d49c324529795ca717ec67fb9178bL279-R281)?

**Comment by mergenci** [comment]
@ulucinar, can you please add `backport release-1.2` label to the PR? Looks like I don't have privileges to do so.

File: pkg/controller/external_tfpluginfw.go [hunk 1]
```diff
@@ -270,13 +271,14 @@ func (n *terraformPluginFrameworkExternalClient) getDiffPlanResponse(ctx context
...
 	}
 
-	// filter diffs that has unknown plan value which corresponds to
-	// computed values. These cause unnecessary diff detection when only computed
-	// attribute diffs exist in the raw diff and no actual diff exists in the
-	// parametrizable attributes
+	// Filter diffs that have unknown plan values, which correspond to
+	// computed fields, and null plan values, which correspond to
+	// optional fields. These cause unnecessary diff detection when only
+	// computed or optional attribute diffs exist in the raw diff and no
```
Comment (ulucinar) [inline pkg/controller/external_tfpluginfw.go:277]:
```suggestion
	// not-specified fields. These cause unnecessary diff detection when only
	// computed attributes or not-specified argument diffs exist in the raw diff and no
```

**Review by ulucinar — APPROVED** [review]
Thanks @mergenci, let's also make sure these changes are consumed in the provider-aws v1.1 branch.

**Review by erhancagirici — APPROVED** [review]
LGTM, thanks @mergenci

**Pull request merged**